cli/src/lib/npm/npmLibDefs.js

```diff
@@ -359,90 +359,128 @@ export async function findNpmLibDef(
 type InstalledNpmLibDef =
   | {|kind: 'LibDef', libDef: NpmLibDef|}
   | {|kind: 'Stub', name: string|};
-export async function getInstalledNpmLibDefs(
-  flowProjectRootDir: string,
-  libdefDir?: string,
-): Promise<Map<string, InstalledNpmLibDef>> {
-  const typedefDir = libdefDir || 'flow-typed';
-  const libDefDirPath = path.join(flowProjectRootDir, typedefDir, 'npm');
-  const installedLibDefs = new Map();
-  if (await fs.exists(libDefDirPath)) {
-    const filesInNpmDir = await getFilesInDir(libDefDirPath, true);
-    await P.all(
-      [...filesInNpmDir].map(async fileName => {
-        const fullFilePath = path.join(libDefDirPath, fileName);
-        const terseFilePath = path.relative(flowProjectRootDir, fullFilePath);
-        const fileStat = await fs.stat(fullFilePath);
-        if (fileStat.isFile()) {
-          const fileContent = (await fs.readFile(fullFilePath)).toString();
-          if (verifySignedCode(fileContent)) {
-            const signedCodeVer = getSignedCodeVersion(fileContent);
-            if (signedCodeVer === null) {
-              return;
-            }
-            const matches = signedCodeVer.match(
-              /([^\/]+)\/(@[^\/]+\/)?([^\/]+)\/([^\/]+)/,
-            );
-            if (matches == null) {
-              return;
-            }
 
-            if (matches[1] === '<<STUB>>') {
-              installedLibDefs.set(terseFilePath, {
-                kind: 'Stub',
-                name: matches[2],
-              });
-              return;
-            }
+type ParsedSignedCodeVersion =
+  | {|
+      kind: 'LibDef',
+      libDef: $Diff<NpmLibDef, {|path: string|}>,
+    |}
+  | {|kind: 'Stub', name: string|};
 
-            const scope =
-              matches[2] == null
-                ? null
-                : matches[2].substr(0, matches[2].length - 1);
+export function parseSignedCodeVersion(
+  signedCodeVer: string,
+): ?ParsedSignedCodeVersion {
+  if (signedCodeVer === null) {
+    return null;
+  }
+  if (signedCodeVer.startsWith('<<STUB>>/')) {
+    return {
+      kind: 'Stub',
+      name: signedCodeVer.substring('<<STUB>>/'.length),
+    };
+  }
```

Comment:
So is moving up this `STUB` check the biggest change here? This diff is kinda hard to read, let me know if there are other changes in this file. Nice refactor by the way. 👌

Reply:
yeah it's the biggest change.  The extracted `getInstalledNpmLibDef` I thought I would need for testing before I realized I only needed to test the `parseSignedCodeVersion` function for now.

Reply:
Cool. With everything else being the same, I'll pull this change locally and test it tomorrow. If all looks good, we'll push out a patch to npm in the short term.